cpp/dolfinx/fem/assemble_matrix_impl.h

```diff
@@ -33,8 +33,9 @@ void assemble_matrix(
     const std::function<int(std::int32_t, const std::int32_t*, std::int32_t,
                             const std::int32_t*, const T*)>& mat_set_values,
     const Form<T>& a, const xtl::span<const T>& constants,
-    const xtl::span<const T>& coeffs, int cstride, const std::vector<bool>& bc0,
-    const std::vector<bool>& bc1);
+    const std::map<std::pair<IntegralType, int>,
+                   std::pair<std::vector<T>, int>>& coefficients,
```

Comment:
I think this will need to use a `span` rather than `std::vector` to avoid forcing the storage type (which would be problematic from Python).